controllers/volumereplicationgroup_webhook.go

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	ramen "github.com/ramendr/ramen/api/v1alpha1"
+	recipe "github.com/ramendr/recipe/api/v1alpha1"
+	authentication "k8s.io/api/authentication/v1"
+	authorization "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+//nolint:lll
+//+kubebuilder:webhook:path=/validate-ramendr-openshift-io-v1alpha1-volumereplicationgroup,mutating=false,failurePolicy=fail,sideEffects=None,groups=ramendr.openshift.io,resources=volumereplicationgroups,verbs=create;update,versions=v1alpha1,name=vvolumereplicationgroup.kb.io,admissionReviewVersions=v1
```

Comment:
As we seem to be catching this on a create, it is feasible for a user to update the recipe or the VRG as desired post creation to add more namespaces to the protection scheme. This would be a violation from a security standpoint afaict.

Reply:
Seems the "write" verbs are only `create` and `update`.  Adding `patch` didn't work:
```sh
unable to install webhooks onto control plane: ValidatingWebhookConfiguration.admissionregistration.k8s.io "validating-webhook-configuration" is invalid: webhooks[3].rules[0].operations[1]: Unsupported value: "patch": supported values: "*", "CONNECT", "CREATE", "DELETE", "UPDATE"
```